Cabal/src/Distribution/Simple/Flag.hs

```diff
@@ -46,18 +46,23 @@ import Prelude ()
 --
 -- 1. list flags eg
 --
--- > --ghc-option=foo --ghc-option=bar
+--   > --ghc-option=foo --ghc-option=bar
 --
--- gives us all the values ["foo", "bar"]
+--   gives us all the values ["foo", "bar"]
 --
 -- 2. singular value flags, eg:
 --
--- > --enable-foo --disable-foo
+--   > --enable-foo --disable-foo
 --
--- gives us Just False
--- So this Flag type is for the latter singular kind of flag.
+--   gives us Just False
+--
+-- So, this Flag type is for the latter singular kind of flag.
 -- Its monoid instance gives us the behaviour where it starts out as
 -- 'NoFlag' and later flags override earlier ones.
+--
+-- In essence, this is only a Maybe with a flipped constructor order,
+-- a missing Ord and Monad instance, and a different Semigroup / Monoid
+-- instance.
```

Comment:
This one I frankly do not find particularly helpful: yes this is isomorphic to `Maybe`, but it is quite common and unremarkable.

The comment stresses two properties: a) switched constructors; b) missing instances. But those two are not remarkable in any way, we could add a Monad instance and switch constructors and everything would run the same, am I wrong.

What `Flag` does is being self-documenting and introducing additional type safety. This is staple in Haskell, so I don't think it deserves a comment.

Reply:
Rereading my comments I have been probably too harsh. A simple “Isomorphic to 'Maybe'” could do (and be useful) here.

Reply:
A reference to Maybe was helpful to me at some point, so I don't mind having it in haddocks. As for "additional type safety": funny enough, phadej mentioned at some point that after all these years he realized that having Maybe there might have been a better choice. I don't remember the details but perhaps the reasoning was about ergonomics of it...

As a casual reader, I'd be intrigued about "different Semi group/Monoid` instance. So, I'd either not mention it at all or (would be my choice) expand a little bit on that.

Reply:
@ulysses4ever I'll keep this in mind for further updates; i.e, it might be useful to expand on how the Semigroup / Monoid works on the top-level description of the type.

And yeah, I was sort of thinking: "Why not a newtype?" Then again, newtypes can be annoying to coerce through...